it/controllers/benefits/assets/AssetsBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,404 @@
+        urlGet(assetsBenefitsPageUrl(taxYear), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+      }
+
+      s"has an SEE OTHER($SEE_OTHER) status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
+      }
+    }
+  }
+
+  ".submit" should {
+    userScenarios.foreach { user =>
+      import Selectors._
+      import user.commonExpectedResults._
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        "return an error when a user submits an empty form" which {
+
+          lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> "")
+
+
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(hasPriorBenefits = true, cyaModel(Some(benefits(Some(fullAssetsModel))))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlPost(assetsBenefitsPageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          s"has a BAD REQUEST($BAD_REQUEST) status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption, captionSelector)
+          textOnPageCheck(user.specificExpectedResults.get.youCanUseText, youCanUseTextSelector)
+          radioButtonCheck(yesText, 1, Some(false))
+          radioButtonCheck(noText, 2, Some(false))
+          buttonCheck(continueText, continueButtonSelector)
+          formPostLinkCheck(formPostLink, formSelector)
+
+          errorSummaryCheck(user.specificExpectedResults.get.expectedErrorText, yesSelector)
+          errorAboveElementCheck(user.specificExpectedResults.get.expectedErrorText, Some("value"))
+        }
+      }
+    }
+
+    "update cya with 'yes' and a user chooses yes and doesn't have prior benefits" which {
+      lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.yes)
+
+      lazy val result: WSResponse = {
+        dropEmploymentDB()
+        authoriseAgentOrIndividual(isAgent = false)
+        insertCyaData(employmentUserData(hasPriorBenefits = false, cyaModel(Some(benefits(Some(
+          fullAssetsModel.copy(assetsQuestion = None)))))), userRequest)
+        urlPost(assetsBenefitsPageUrl(taxYearEOY), body = form, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "redirect to the assets amount page" in {
+        result.status shouldBe SEE_OTHER
+        //TODO - change to assets amount page when available
```

Comment:
This page is already in master. Can you please update that?

Reply:
as discussed this will be left for the navigation ticket